textproc/tesseract/Portfile

```diff
@@ -4,19 +4,33 @@ PortSystem          1.0
 PortGroup           github 1.0
 
 if {${subport} in [list tesseract tesseract-training]} {
-    github.setup    tesseract-ocr tesseract 4.1.1
-    checksums       rmd160  2dd932d66b7271443ecce27359459f319a1af1a8 \
-                    sha256  5e477a1219c8a7e0dde16d2452edcd978525bea7b9cb9e1a9b3cc78e603c454b \
-                    size    1975412
-} else {
-    github.setup    tesseract-ocr tessdata  4.0.0
-    revision        0
-    checksums       rmd160  037e33f19787c2f08544bcde213eb202f73d9546 \
-                    sha256  1c36d1e521e6420ea9c8bf13d8a45f67d54fdcf6a132fa72370ae85464118b96 \
-                    size    669255785
+    github.setup    tesseract-ocr tesseract 4.1.3
+    checksums       rmd160  e2037d64e5549ed0de715ef2d2b082149f3f9232 \
+                    sha256  26ea123135c3e0b4f204cd097a9d43ab4293f23425e96128df48e6da1960b40f \
+                    size    1975794
+} elseif {[variant_isset best]} {
```

Comment:
The `subport` error is due to the structure of your `if`/`elseif` logic: Note that you have no fall-through `else` case, and you never unconditionally set the port name anywhere.

So for the default port, with no variants selected, you never set `name`.

Reply:
Actually you need to do more than just set `name` in your fallthrough logic, you also need to call `github.setup` appropriately as well.

Reply:
So the $100,000 question is: Do you no longer want the default port (sans any variants selected by the user) to install something?

That fundamental question will drive what you need to change/fix.

Reply:
That being said, you should really ensure that the default port provides the same functionality as it did before your update, rather than yanking the rug out from under the user.

Reply:
So I'd suggest eliminating the `legacy` variant, and use that as the default case.

You can still call out the fact that it's now legacy, though, via the port notes.

Reply:
Also, please eliminate the copy-pasta where you set the name to `tesseract` in the various `if`/`elseif` blocks, and simply set that at the top of the portfile. (On line 6, with a blank line between that and your selection logic.)

Reply:
p.s. Not trying to be a pain here, just trying to help you streamline and simplify. And does all of that make sense?

Reply:
No, thanks a lot - I was having trouble with streamlining this.